src/service.rs

```diff
@@ -0,0 +1,278 @@
+    /// Restart the service.
+    Restart(StandardOpts),
+    /// Print the status of the service.
+    Status(StandardOpts),
+}
+
+struct ServiceInfo {
+    pub name: OsString,
+    pub display_name: OsString,
+    pub description: OsString,
+
+    pub executable_path: std::path::PathBuf,
+    pub launch_arguments: Vec<OsString>,
+}
+
+impl Default for ServiceInfo {
+    fn default() -> Self {
+        let current_exe = ::std::env::current_exe().unwrap();
+
+        ServiceInfo {
+            name: OsString::from(*DEFAULT_SERVICE_NAME),
+            display_name: OsString::from("Vector Service"),
+            description: OsString::from(built_info::PKG_DESCRIPTION),
+            executable_path: current_exe,
+            launch_arguments: vec![],
+        }
+    }
+}
+
+#[derive(Debug, Clone, PartialEq)]
+enum ControlAction {
+    Install,
+    Uninstall,
+    Start,
+    Stop,
+    Restart,
+    Status,
+}
+
+pub fn cmd(opts: &Opts) -> exitcode::ExitCode {
+    let sub_command = &opts.sub_command;
+    let code = match sub_command {
+        Some(s) => match s {
+            SubCommand::Install(opts) => {
+                control_service(&opts.service_info(), ControlAction::Install)
+            }
+            SubCommand::Uninstall(opts) => {
+                control_service(&opts.service_info(), ControlAction::Uninstall)
+            }
+            SubCommand::Start(opts) => control_service(&opts.service_info(), ControlAction::Start),
+            SubCommand::Stop(opts) => control_service(&opts.service_info(), ControlAction::Stop),
+            SubCommand::Restart(opts) => {
+                control_service(&opts.service_info(), ControlAction::Restart)
+            }
+            SubCommand::Status(opts) => {
+                control_service(&opts.service_info(), ControlAction::Status)
+            }
+        },
+        None => {
+            let default_service = ServiceInfo::default();
+            control_service(&default_service, ControlAction::Status)
+        }
+    };
+
+    code
+}
+
+#[cfg(windows)]
+fn print_status(service_info: &ServiceInfo) -> windows_service::Result<()> {
+    use windows_service::{
+        service::ServiceAccess,
+        service_manager::{ServiceManager, ServiceManagerAccess},
+    };
+
+    let service_name = &service_info.name;
+
+    let manager_access = ServiceManagerAccess::CONNECT;
+    let manager = ServiceManager::local_computer(None::<&str>, manager_access)?;
+
+    let service_access = ServiceAccess::QUERY_CONFIG | ServiceAccess::QUERY_STATUS;
+    let service = manager.open_service(service_name, service_access)?;
+
+    let status = service.query_status()?;
+    let config = service.query_config()?;
+
+    let name_string = format!("{:?}", service_name);
+    let state_string = format!("{:?}", status.current_state);
+
+    let state_width = state_string.len() + 2;
+    let name_width = name_string.len() + 2;
+
+    println!(
+        "{0: <state_width$} {1: <name_width$} Display Name",
```

Comment:
Based on this, I would expect more output from this command? 
![image](https://user-images.githubusercontent.com/130903/90795223-a4860200-e2c2-11ea-8c91-0fae17c29ea2.png)